src/components/containers/UpOnlyWBTC.tsx

```diff
@@ -20,16 +20,32 @@ const UpOnlyWBTC = () => {
         currentPrice,
         expiry,
       } = await stakedPut.methods.getInputs(purchaseAmount).call();
-      // TODO: Call buyInstrument with inputs
-      console.log(
-        `wbtcSize: ${wbtcSize}`,
-        `expDigg: ${expDigg}`,
-        `tradeAmt: ${tradeAmt}`,
-        `premium: ${premium}`,
-        `totalCost: ${totalCost}`,
-        `currentPrice: ${currentPrice}`,
-        `expiry: ${expiry}`
-      );
+      const encodedBuyInstrumentCall = await (
+        await stakedPut.methods.buyInstrument([
+          process.env.REACT_APP_NODE_ENV ? "5000000000000" : currentPrice,
+          wbtcSize,
+          process.env.REACT_APP_NODE_ENV
+            ? "1000000000000000000000"
+            : premium + 1,
+          expiry,
+          purchaseAmount,
+          tradeAmt,
+          wbtcSize,
+          expDigg,
+        ])
+      ).encodeABI();
+
+      await window.ethereum.request({
+        method: "eth_sendTransaction",
+        params: [
+          {
+            to: stakedPut._address,
+            from: window.ethereum.selectedAddress,
+            value: totalCost,
+            data: encodedBuyInstrumentCall,
+          },
+        ],
```

Comment:
Instead of calling `eth_sendTransaction` in the raw way, we can actually invoke the `buyInstrument` method. See:

```js
await stakedPut.methods.buyInstrument([
          process.env.REACT_APP_NODE_ENV ? "5000000000000" : currentPrice,
          wbtcSize,
          process.env.REACT_APP_NODE_ENV
            ? "1000000000000000000000"
            : premium + 1,
          expiry,
          purchaseAmount,
          tradeAmt,
          wbtcSize,
          expDigg,
        ]).send({
            to: stakedPut._address,
            from: window.ethereum.selectedAddress,
            value: totalCost,
          })
```

Reply:
Were you able to successfully do so? It doesn't seem to trigger a MetaMask transaction and also results in this error:
`Error: Returned error: The method eth_sendTransaction does not exist/is not available`.

Reply:
Hmm this should have worked... https://web3js.readthedocs.io/en/v1.2.11/web3-eth-contract.html#methods-mymethod-send

Reply:
May be because I’m using the web3-eth-contract package. I’m going to keep it as is, unless it’s urgent, and someone else can come along and use the full web3 package if necessary.

Reply:
Is send being called on the vault frontend? Curious if MetaMask would pick up on the call and the “send tx” pop up would appear. I haven’t used the ribbon protocol yet so I’m not sure (will do so soon).